src/main/java/uk/gov/digital/ho/hocs/queue/ukvi/UKVIComplaintValidator.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.digital.ho.hocs.queue.ukvi;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.util.ResourceUtils;
+import uk.gov.digital.ho.hocs.application.ClientContext;
+import uk.gov.digital.ho.hocs.client.audit.AuditClient;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Set;
+
+@Slf4j
+@Service
+public class UKVIComplaintValidator {
+
+    private final ObjectMapper objectMapper;
+    private final JsonSchema schema;
+    private final UKVITypeData ukviTypeData;
+    private final AuditClient auditClient;
+    private final ClientContext clientContext;
+    private String user;
+    private String group;
+
+    @Autowired
+    public UKVIComplaintValidator(ObjectMapper objectMapper,
+                                  UKVITypeData ukviTypeData,
+                                  AuditClient auditClient,
+                                  ClientContext clientContext,
+                                  @Value("${case.creator.ukvi-complaint.user}") String user,
+                                  @Value("${case.creator.ukvi-complaint.group}") String group) throws IOException {
+        this.ukviTypeData = ukviTypeData;
+        this.auditClient = auditClient;
+        this.clientContext = clientContext;
+        this.user = user;
+        this.group = group;
+        File file = ResourceUtils.getFile("classpath:cmsSchema.json");
+        JsonSchemaFactory schemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V4);
+        schema = schemaFactory.getSchema(file.toURI());
+        this.objectMapper = objectMapper;
+    }
+
+    public void validate(String jsonBody, String messageId) throws Exception {
+        clientContext.setContext(user, group, messageId);
+        JsonNode json = objectMapper.readTree(jsonBody);
+        Set<ValidationMessage> validationMessages = schema.validate(json);
+        if (validationMessages.isEmpty()) {
+            auditClient.audit(ukviTypeData.getSuccessfulValidationEvent(), null, null, jsonBody);
```

Comment:
"null, null" here will leave "caseType" blank, so these won't show in any audit extracts. Are there any MI requirements around this?

Reply:
I'll remove this from here, and add the payload to the caae_created event.
Does that sound ok?

Reply:
I made up the requirements!

Reply:
:-)